theseus/core/tests/test_objective.py

```diff
@@ -292,8 +292,8 @@ def _check_error_for_data(v1_data_, v2_data_, error_, error_norm_2_):
         )
 
         # To check if variables are not updated
-        assert not torch.equal(objective.optim_vars["v1"].data, input_data["v1"])
-        assert not torch.equal(objective.optim_vars["v2"].data, input_data["v2"])
+        assert objective.optim_vars["v1"].data is not input_data["v1"]
```

Comment:
In addition to this check, I still think it's worth it to also confirm that the previous tensor is still there. It's a bit conservative, but it might turn up to be useful in the future, and it's not too much extra effort.

Reply:
@luisenp like to store the variables before update like `old_data = objective.optim_vars` and checking after `assert old_data is objective.optim_vars`

Reply:
Mostly why I outlined in [this](https://github.com/facebookresearch/theseus/pull/105#discussion_r823873662) comment.

Reply:
@luisenp Updated the test